code/modules/projectiles/guns/chaos_bolt.dm

```diff
@@ -0,0 +1,348 @@
+#define CHAOS_STAFF_DAMAGE 30 //Damaging effects will use this number, multiplied or divided depending on effect
+#define CHAOS_STAFF_LETHAL_CHANCE 5 //These should add up to 100
+#define CHAOS_STAFF_NEGATIVE_CHANCE 45
+#define CHAOS_STAFF_MISC_CHANCE 30
+#define CHAOS_STAFF_GIFT_CHANCE 15
+#define CHAOS_STAFF_GREAT_GIFT_CHANCE 5
+
+/obj/item/projectile/magic/chaos
+	name = "chaos bolt"
+	icon_state = "ice_1"
+	impact_effect_type = /obj/effect/temp_visual/impact_effect/chaos
+	var/obj/item/item_to_summon
+	/// If left at 0, item goes in backpack or floor, if set, throw that many items around the target
+	var/explosion_amount = 0
+	var/chaos_effect
+
+/obj/item/projectile/magic/chaos/on_hit(atom/target, blocked = 0)
+	. = ..()
+
+	if(iswallturf(target) || isobj(target))
+		target.color = pick(GLOB.random_color_list)
```

Comment:
```suggestion
	if(iswallturf(target) || isobj(target))
		target.color = pick(GLOB.random_color_list)
		return
```
just return here since the next if check is guaranteed to fail